pkg/tenant/write.go

```diff
@@ -30,66 +33,77 @@ func CreateOrUpdate(
 ) (result CreateOrUpdateResult, err error) {
 	result = CreateOrUpdateResult{}
 
-	repository, err := git.OpenAndResetRepositoryState(op.CplatformRepoPath)
+	repository, err := git.OpenAndResetRepositoryState(op.CplatformRepoPath, op.DryRun)
 	if err != nil {
 		return result, fmt.Errorf("couldn't open cplatform repository: %v", err)
 	}
 
 	if err = repository.CheckoutBranch(&git.CheckoutOp{
 		BranchName:      op.BranchName,
 		CreateIfMissing: true,
+		DryRun:          op.DryRun,
 	}); err != nil {
 		return result, err
 	}
 	defer func() {
-		_ = repository.CheckoutBranch(&git.CheckoutOp{BranchName: git.MainBranch})
+		_ = repository.CheckoutBranch(&git.CheckoutOp{BranchName: git.MainBranch, DryRun: op.DryRun})
 	}()
 
-	if err = tenant.CreateOrUpdate(tenant.CreateOrUpdateOp{
-		Tenant:       op.Tenant,
-		ParentTenant: op.ParentTenant,
-		TenantsDir:   tenant.DirFromCPlatformPath(op.CplatformRepoPath),
-	}); err != nil {
-		return CreateOrUpdateResult{}, err
-	}
+	log.Debug().Msg("writing tenant definition to cplatform repo")
```

Comment:
Should we do a separate log message for each operation in case of dryRun? It also might be useful to log the resulting tenant and the path to save it